addons/redis/scripts/redis-sentinel-offline.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+set -ex
+
+declare -g redis_default_service_port=26379
+declare -A master_slave_counts
+
+member_leave_sentinel() {
+  if [ -z "$KB_LEAVE_MEMBER_POD_IP" ]; then
+    echo "Error: Required environment variable KB_LEAVE_MEMBER_POD_IP is not set."
+    exit 1
+  fi
+
+  if [ -z "$KB_LEAVE_MEMBER_POD_NAME" ]; then
+    echo "Error: Required environment variable KB_LEAVE_MEMBER_POD_NAME is not set."
+    exit 1
+  fi
+
+  if [ -z "$KB_MEMBER_ADDRESSES" ]; then
+    echo "Error: Required environment variable KB_MEMBER_ADDRESSES is not set."
+    exit 1
+  fi
+
+  sentinel_leave_member_name=$KB_LEAVE_MEMBER_POD_NAME
+  sentinel_leave_member_ip=$KB_LEAVE_MEMBER_POD_IP
+
+  old_ifs="$IFS"
+  IFS=','
+  set -f
+  read -ra sentinel_pod_list <<< "${KB_MEMBER_ADDRESSES}"
+  set +f
+  IFS="$old_ifs"
+
+  touch /data/sentinel/member_leave.conf
```

Comment:
The approach of using `touch /data/sentinel/member_leave.conf` has some issues. This file will remain on the pvc and will be backed up. Imagine a scenario where a user restores an instance from a backup; the presence of this member leave file could prevent it from starting properly.